src/Insect/Parser.purs

```diff
@@ -89,10 +89,10 @@ whiteSpace = token.whiteSpace
 -- | Parse a number.
 number ∷ P Decimal
 number = do
-  intPart ← digits
-
-  mFracPart ← optionMaybe (append <$> string "." <*> digits)
-  let fracPart = fromMaybe "" mFracPart
+  decimalPart ← fractionalPart <|> do
```

Comment:
Suggestions on how to better implement this are welcome, I feel like I could do this better.